src/Documenter.jl

```diff
@@ -441,4 +445,93 @@ end
 
 end
 
+"""
+    generate(
+        pkgname;
+        dir = "<package directory>"
+    )
+
+Creates a documentation stub for a package called `pkgname`. The location of
+the documentation is assumed to be `<package directory>/docs`, but this can
+be overriden with keyword arguments.
+
+It creates the following files
+
+    docs/
+        .gitignore
+        src/index.md
+        make.jl
+        mkdocs.yml
+
+# Positionals
+
+**`pkgname`** is the name of the package (without `.jl`). It is used to
+determine the location of the documentation if `dir` is not provided.
+
+# Keywords
+
+**`dir`** defines the directory where the documentation will be generated.
+It defaults to `<package directory>/docs`. The directory must not exist.
+
+# Examples
+
+    julia> using Documenter
+
+    julia> Documenter.generate("MyPackageName")
+    [ ... output ... ]
+
+"""
+function generate(pkgname::AbstractString; dir=nothing)
+    # TODO:
+    #   - `format` argument to determine the deployment (e.g. mkdocs)
+    #   - set up deployment to `gh-pages`?
+    #   - clean up if it fails
+
+    # Check the validity of the package name
+    if length(pkgname) == 0
+        error("Package name can not be an empty string.")
+    end
+    # Determine the root directory where we wish to generate the docs and
+    # check that it is a valid directory.
+    docroot = if dir == nothing
+        pkgdir = Pkg.dir(pkgname)
+        if !isdir(pkgdir)
+            error("Unable to find package $(pkgname).jl at $(pkgdir).")
+        end
+        docdir = joinpath(pkgdir, "docs")
+        if ispath(docdir)
+            error("$(dir) already exists.")
+        end
+        docdir
+    else
+        if ispath(dir)
+            error("Directory $(dir) already exists.")
+        end
+        dir
+    end
+
+    # deploy the stub
+    info("Deploying documentation to $(docroot)")
+    mkdir(docroot)
+
+    # create the root doc files
+    Generator.savefile(docroot, ".gitignore") do io
+        write(io, Generator.gitignore())
+    end
+    Generator.savefile(docroot, "make.jl") do io
+        write(io, Generator.make(pkgname))
+    end
+    Generator.savefile(docroot, "mkdocs.yml") do io
+        write(io, Generator.mkdocs(pkgname))
+    end
+
+    # Create the default documentation source files
+    mkdir(joinpath(docroot,"src"))
```

Comment:
It would make the code a bit nicer I think. One option would be to call as `savefile(root,"src/index.md")` and then have `savefile` split the name on `/`s and create the directories, if necessary.